server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -407,6 +407,55 @@ public boolean removeSegment(SegmentId segmentId)
     }
   }
 
+  @Override
+  public boolean disableSegments(String dataSource, Collection<String> segmentIds)
+  {
+    if (segmentIds.isEmpty()) {
+      return false;
+    }
+    try {
+      connector.getDBI().withHandle(handle -> {
+        Batch batch = handle.createBatch();
+        segmentIds
+            .forEach(segmentId -> batch.add(
+                StringUtils.format(
+                    "UPDATE %s SET used=false WHERE id = '%s'",
+                    getSegmentsTable(),
+                    segmentId
+                )
+            ));
+        return batch.execute();
```

Comment:
batch.execute() returns a int[] indicating the result of each individual statement, I think it would be more informative if the response for the segmentIds API told the user per-segment success/fail

Reply:
hmm, should it be just `OK` if all segments passed and `FAIL` if all failed. And return per-segment success/failure only of there are mixed results. I am thinking what if there's huge number of segments in the interval, does it make sense to return just the number of segments successfully marked unused ?

Reply:
For the interval case, I think returning a count of segments marked unused makes sense (could provide a non-default option to report exactly what was dropped).

For the segmentId case, I think it makes more sense to report individual failures (if the user is trying to enable specific segments, then I think it's likely that they'll care about the specific results). I think it's fine to leave out successes and report segments that were not successfully disabled.

Reply:
@jon-wei @surekhasaharan - since we're talking about design level stuff now, let's please focus on that before going back to code level comments. When it's sorted out could you update please take this discussion to https://github.com/apache/incubator-druid/issues/7439, and flesh out the proposal to include what the the response entails? Let's try to keep the PR focused on code level things.

My 2¢ is that the response format should be the same regardless of whether you're doing by interval or by segment (for consistency's sake, so if people switch from one to the other they don't get surprised). And probably it's fine to just return the number of segments that had their status changed.

If a user specifies some specific segments that don't exist, I'd be inclined to error out the entire call. But am not 100% tied to this.

Reply:
Also what if for some reason we fail to update 1 of the segments passed. Should it rollback or just return segments-1 for modified?

Reply:
I think it's fine to return segments-1, I don't think this API needs to provide atomic updates

Reply:
But it should return some kind of error code too. The caller needs to know that not everything they asked for was done.